CMakeLists.txt

```diff
@@ -33,7 +33,25 @@ if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
     message(FATAL_ERROR "Do not build in-source.\nPlease remove CMakeCache.txt and the CMakeFiles/ directory.\nThen: mkdir build ; cd build ; cmake .. ; make")
 endif()
 
+# Backwards Compatibility
+# TODO: remove this once there has been a period to enable people to migrate
+set(_CMARK_BUILD_SHARED_LIBS_DEFAULT NO)
+if(CMARK_SHARED)
```

Comment:
This test doesn't work if CMARK_SHARED is set to OFF/NO/FALSE. We should check whether the variable is defined.

Reply:
Great catch! Fixed.